package/bin/irc/irc.js

```diff
@@ -231,8 +231,15 @@
           var line = this.data.slice(0, cr ? crlf - 1 : crlf);
           this.data = this.data.slice(crlf + 1);
           dataView = new Uint8Array(this.data);
-          _results.push(this.util.fromSocketData(line, function(lineStr) {
-            console.log('<=', "(" + _this.server + ")", lineStr);
+          _results.push(this.util.fromSocketData(line, function(lineStr, encoding) {
+            console.groupCollapsed('<=', "(" + _this.server + ")", lineStr);
+            console.log(
+              encoding,
+              lineStr.length + ' characters',
+              line.byteLength + ' bytes'
+            );
+            _this.util.dumpBuffer(line);
+            console.groupEnd();
             return _this.onServerMessage(_this.util.parseCommand(lineStr));
           }));
```

Comment:
That's only necessary if this is actually being called after the loop finishes referring to variables updated or defined in the loop. If it's called immediately (in fromSocketData) it doesn't matter.